### PR TITLE
moved entities from area manager to use case

### PR DESCRIPTION
**Pull request by mahirhassan-uoft**
- fixed some text about "You decided to no" :skull:
- added clean architecture stuff to eventManager for clean architecture
- implemented textSpeed to dialogue

### CODE REVIEW

File: src/main/java/game_world/managers/EventDatabaseInteractor.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     * Generates new event object
+     * @return the new event generated
+     * @param name name of Event to be searched
+     * @param exec adds Event to eventQueue to be executed if set to true
+     */
+    public Event createEvent(ArrayList<Event> eventQueue, String name, boolean exec) {
+        String type = this.database.fetchEventType(name);
+        Event newEvent;
+
+        if (type.equals("Encounter")) {
```
Comment (bw55555) [inline src/main/java/game_world/managers/EventDatabaseInteractor.java:33]: should move this method into EventFactory maybe? Makes more sense imo

File: src/main/java/game_world/managers/EventDatabaseInteractor.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        this.eventFactory = new EventFactory();
+    }
+
+    /**
+     * Generates new event object
+     * @return the new event generated
+     * @param name name of Event to be searched
+     * @param exec adds Event to eventQueue to be executed if set to true
+     */
+    public Event createEvent(ArrayList<Event> eventQueue, String name, boolean exec) {
```
Comment (bw55555) [inline src/main/java/game_world/managers/EventDatabaseInteractor.java:29]: what is the eventQueue parameter for?

**Review by bw55555 — CHANGES_REQUESTED** [review]

File: src/main/java/game_world/managers/EventDatabaseInteractor.java (hunk 1 above)
Comment (bw55555) [reply src/main/java/game_world/managers/EventDatabaseInteractor.java:33]: never mind, this is probably fine

**Review by Yasutakahi — APPROVED** [review]
I think the changes make sense and the documentation looks good to me!

**Review by passionfruitjello — APPROVED** [review]
looks fine

**Review by bw55555 — APPROVED** [review]
Looks good, no entity code left in use case!

**Review by juhwansonn — APPROVED** [review]

**Pull request merged**